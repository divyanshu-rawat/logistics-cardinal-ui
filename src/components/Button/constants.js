export const SIZES = {
  XSMALL: 'xsmall',
  SMALL: 'small',
  DEFAULT: 'default',
  LARGE: 'large',
};

export const ACTION_TYPES = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning',
  DANGER: 'danger',
  LINK: 'link',
};
