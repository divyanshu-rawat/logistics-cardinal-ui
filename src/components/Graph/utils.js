/**
 * Generate random shifts data
 * @returns {{shifts: number}[]}
 */
export function generateRandomShifts() {
  const MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
  return [...new Array(12)].map((_, index) => ({
    shifts: Math.round(Math.random() * 50),
    month: MONTHS[index],
  }));
}
