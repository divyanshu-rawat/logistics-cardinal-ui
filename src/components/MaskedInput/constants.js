import createNumberMask from 'text-mask-addons/dist/createNumberMask';
import emailMask from 'text-mask-addons/dist/emailMask';

// DD.MM.YYYY
export const BIRTHDAY_MASK = [
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  /\d/,
  /\d/,
];

// HH:mm
export const TIME_MASK = [/[0-9]/, /[0-9]/, ':', /[0-9]/, /[0-9]/];

// 10 % || - 10 %
export const PERCENTAGE_MASK = createNumberMask({
  prefix: '',
  includeThousandsSeparator: false,
  suffix: ' %',
  allowNegative: true,
});

// $ 10
export const MONEY_MASK = createNumberMask();

// rick@c137.com
export const EMAIL_MASK = emailMask;

// 10.29
export const DECIMAL_MASK = createNumberMask({
  prefix: '',
  allowDecimal: true,
  requireDecimal: true,
});

export default {
  BIRTHDAY_MASK,
  MONEY_MASK,
  EMAIL_MASK,
  DECIMAL_MASK,
  PERCENTAGE_MASK,
};
