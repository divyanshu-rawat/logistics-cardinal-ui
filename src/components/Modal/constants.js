export const modalSizes = {
  small: 'small',
  default: 'default',
};

export const modalClassName = 'CardinalModal__Content';
export const modalOverlayClassName = 'CardinalModal__Overlay';
