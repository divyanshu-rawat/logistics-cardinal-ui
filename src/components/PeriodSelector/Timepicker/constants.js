const theme = {
  brandPrimary: '#3f9bfd',
  brandPrimaryLight: '#375f8a',
  greyMedium: '#343a42',
  greyDeep: '#292d33',
  greyLight: '#3f454d',
  inactive: '#96a2b0',
};

export default theme;
