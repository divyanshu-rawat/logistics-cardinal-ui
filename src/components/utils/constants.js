export const SIZES = {
  lg: 'lg',
  md: 'md',
  sm: 'sm',
  xs: 'xs',
};
