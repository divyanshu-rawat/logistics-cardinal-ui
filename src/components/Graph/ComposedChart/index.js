import React from 'react';
import PropTypes from 'prop-types';
import { ComposedChart as RCComposedChart } from 'recharts';

import { SIZES } from '../../utils/constants';

export function getMarginPixels(size) {
  let pixels;

  switch (size) {
    case SIZES.lg:
      pixels = 30;
      break;
    case SIZES.md:
      pixels = 20;
      break;
    case SIZES.sm:
      pixels = 10;
      break;
    default:
      pixels = 0;
  }

  return { top: pixels, right: pixels, left: pixels, bottom: pixels };
}

function ComposedChart({
  children,
  marginSize,
  data,
  height,
  width,
  ...props
}) {
  const margin = getMarginPixels(marginSize);

  return (
    <RCComposedChart
      margin={margin}
      data={data}
      height={height}
      width={width}
      {...props}
    >
      {children}
    </RCComposedChart>
  );
}

ComposedChart.propTypes = {
  /**
   * Components placed inside the ComposedChart
   */
  children: PropTypes.node.isRequired,
  /**
   * Size of margins applied to four sides of the ComposedChart
   */
  marginSize: PropTypes.oneOf(Object.values(SIZES)).isRequired,
  /**
   * Data passed to children components
   */
  data: PropTypes.array.isRequired,
  /**
   * Height of the component
   */
  height: PropTypes.number.isRequired,
  /**
   * Width of the component
   */
  width: PropTypes.number.isRequired,
};

ComposedChart.defaultProps = {
  marginSize: SIZES.xs,
  height: 300,
  width: 1000,
};

export default ComposedChart;
