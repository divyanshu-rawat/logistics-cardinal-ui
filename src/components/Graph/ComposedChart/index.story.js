import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { select } from '@storybook/addon-knobs';
import { CartesianGrid, Legend, XAxis, YAxis } from 'recharts';

import withTests from '../../../utils/withTests';
import { GRAPH } from '../../../../.storybook/sections';

import Line from '../Line';
import ComposedChart from '.';
import { generateRandomShifts } from '../utils';
import { SIZES } from '../../utils/constants';

storiesOf(`${GRAPH}ComposedChart`, module)
  .addDecorator(withTests('ComposedChart/tests/index'))
  .add(
    'Simple',
    withInfo()(() => (
      <ComposedChart
        marginSize={select('marginSize', Object.values(SIZES), SIZES.xs)}
        data={generateRandomShifts()}
      >
        <XAxis dataKey="month" />
        <YAxis />
        <CartesianGrid strokeDasharray="3 3" />
        <Legend />
        <Line dataKey="shifts" />
      </ComposedChart>
    )),
  );
