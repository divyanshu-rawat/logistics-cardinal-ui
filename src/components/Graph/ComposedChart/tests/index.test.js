import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ComposedChart, { getMarginPixels } from '..';
import { SIZES } from '../../../utils/constants';

const renderedComponent = () =>
  shallow(
    <ComposedChart marginSize="lg" data={[]}>
      Awesome chart
    </ComposedChart>,
  );

function generateMargins(size) {
  return { top: size, right: size, left: size, bottom: size };
}

describe('<ComposedChart />', () => {
  it('should render the component', () => {
    const tree = toJson(renderedComponent());
    expect(tree).toMatchSnapshot();
  });

  it('should get correct margin pixels', () => {
    expect(getMarginPixels(SIZES.lg)).toEqual(generateMargins(30));
    expect(getMarginPixels(SIZES.md)).toEqual(generateMargins(20));
    expect(getMarginPixels(SIZES.sm)).toEqual(generateMargins(10));
    expect(getMarginPixels(SIZES.xs)).toEqual(generateMargins(0));
  });
});
