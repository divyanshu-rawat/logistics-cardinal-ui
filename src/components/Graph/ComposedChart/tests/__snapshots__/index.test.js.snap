// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposedChart /> should render the component 1`] = `
<ComposedChart
  barCategoryGap="10%"
  barGap={4}
  data={Array []}
  height={300}
  layout="horizontal"
  margin={
    Object {
      "bottom": 30,
      "left": 30,
      "right": 30,
      "top": 30,
    }
  }
  reverseStackOrder={false}
  stackOffset="none"
  width={1000}
>
  Awesome chart
</ComposedChart>
`;
